applications/crossbar/src/modules/cb_push_notification_subscriptions.erl

```diff
@@ -0,0 +1,215 @@
+read_docs_for_app_and_mobile_device_id(Context, App, MobileDeviceId) ->
+    lager:debug("Reading push notification subscription: ~p - ~p", [App, MobileDeviceId]),
+    case subscription_does_exist(App, MobileDeviceId, Context) of
+        {'exists', Data} ->
+            lager:debug("found document for push notification: ~p", [Data]),
+            NewContext = cb_context:set_doc(Context, Data),
+            crossbar_util:response(Data, NewContext);
+        {_Status, BadContext} ->
+            lager:debug("Could not read document for push notifications"),
+            BadContext
+    end.
+
+%%------------------------------------------------------------------------------
+%% @doc Called when request has validated successfully, prepares document for saving in db
+%% @end
+%%------------------------------------------------------------------------------
+-spec on_successful_validation(kz_term:api_ne_binary(), kz_term:ne_binary(), kz_term:ne_binary(), cb_context:context()) -> cb_context:context().
+on_successful_validation('undefined', App, MobileDeviceId, Context) ->
+    %% Set values that came from path
+    lager:debug("New push notification subscription validated successfully"),
+    NewDoc = lists:foldl(fun(F, Doc) -> F(Doc) end, cb_context:doc(Context),
+                         [fun(X) -> kz_json:set_value(<<"pvt_user_id">>, cb_context:auth_user_id(Context), X) end
+                         ,fun(X) -> kz_json:set_value(<<"pvt_app_name">>, App, X) end
+                         ,fun(X) -> kz_json:set_value(<<"pvt_device_id">>, MobileDeviceId, X) end
+                         ,fun(X) -> kz_doc:set_type(X, <<"push_notification_subscription">>) end
+                         ]),
+    lager:debug("Set push notification subscription doc filled with private values: ~p", [NewDoc]),
+    cb_context:set_doc(Context, NewDoc).
+-spec on_successful_validation(kz_term:api_ne_binary(), cb_context:context()) -> cb_context:context().
+on_successful_validation(Id, Context) ->
+    crossbar_doc:load_merge(Id, Context, ?TYPE_CHECK_OPTION(<<"push_notification_subscription">>)).
+
+%%------------------------------------------------------------------------------
+%% @doc Normalizes the results of a view
+%% @end
+%%------------------------------------------------------------------------------
+-spec normalize_view_results(kz_json:object(), kz_json:objects()) -> kz_json:objects().
+normalize_view_results(JObj, Acc) ->
+    [kz_json:get_json_value(<<"value">>, JObj)|Acc].
+
+-spec subscription_does_exist(kz_term:ne_binary(), kz_term:ne_binary(), cb_context:context()) -> {'not_exists' | 'error', cb_context:context()} | {'exists', kz_term:api_object()}.
+subscription_does_exist(App, MobileDeviceId, Context) ->
+    ViewContext = crossbar_doc:load_view(?BY_APP_BY_DEVICE, [{'key', [cb_context:auth_user_id(Context), MobileDeviceId, App]}], Context, fun normalize_view_results/2),
```

Comment:
Look into using `crossbar_view` and maybe that will address the issue for why you're not able to use `crossbar_doc:delete`. It seems awfully strange that `delete` didn't work for you and I would rather fix how Context is getting loaded on the read portion.

Reply:
Thanks for the tip. I'll try fix it! :)